python/cudf/cudf/_libxx/parquet.pyx

```diff
@@ -101,61 +97,35 @@ cpdef read_parquet(filepath_or_buffer, columns=None, row_group=None,
-    options.strings_to_categorical = strings_to_categorical
-    options.use_pandas_metadata = use_pandas_metadata
+        args.columns.push_back(str(col).encode())
+    args.strings_to_categorical = strings_to_categorical
+    args.use_pandas_metadata = use_pandas_metadata
 
-    # Create reader from source
-    cdef const unsigned char[::1] buffer = lib.view_of_buffer(
-        filepath_or_buffer)
-    cdef string filepath
-    if buffer is None:
-        if not os.path.isfile(filepath_or_buffer):
-            raise FileNotFoundError(
-                errno.ENOENT, os.strerror(errno.ENOENT), filepath_or_buffer
-            )
-        filepath = <string>str(filepath_or_buffer).encode()
-
-    cdef unique_ptr[parquet_reader] reader
-    with nogil:
-        if buffer is None:
-            reader = unique_ptr[parquet_reader](
-                new parquet_reader(filepath, options)
-            )
-        else:
-            reader = unique_ptr[parquet_reader](
-                new parquet_reader(<char*>&buffer[0], buffer.shape[0], options)
-            )
-
-    # Read data into columns
-    cdef cudf_table c_out_table
-    cdef size_type c_skip_rows = skip_rows if skip_rows is not None else 0
-    cdef size_type c_num_rows = num_rows if num_rows is not None else -1
-    cdef size_type c_row_group = row_group if row_group is not None else -1
-    with nogil:
-        if c_skip_rows != 0 or c_num_rows != -1:
-            c_out_table = reader.get().read_rows(c_skip_rows, c_num_rows)
-        elif c_row_group != -1:
-            c_out_table = reader.get().read_row_group(c_row_group)
-        else:
-            c_out_table = reader.get().read_all()
+    args.skip_rows = skip_rows if skip_rows is not None else 0
+    args.num_rows = num_rows if num_rows is not None else -1
+    args.row_group = row_group if row_group is not None else -1
+    args.row_group_count = row_group_count \
+        if row_group_count is not None else -1
+    args.timestamp_type = cudf_types.data_type(cudf_types.type_id.EMPTY)
 
-    # Construct dataframe from columns
-    df = table_to_dataframe(&c_out_table)
+    # Read Parquet
+    cdef cudf_io_types.table_with_metadata c_out_table
 
-    # Set column to use as row indexes if available
-    index_col = reader.get().get_index_column().decode("UTF-8")
-    if index_col is not '' and index_col in df.columns:
-        df = df.set_index(index_col)
-        new_index_name = pa.pandas_compat._backwards_compatible_index_name(
-            df.index.name, df.index.name
-        )
-        df.index.name = new_index_name
+    with nogil:
+        c_out_table = move(parquet_reader(args))
 
-    return df
+    column_names = list(c_out_table.metadata.column_names)
+    column_names = [x.decode() for x in column_names]
+    cdef unique_ptr[table] ctbl = move(c_out_table.tbl)
+    return Table.from_unique_ptr(move(c_out_table.tbl),
+                                 column_names=column_names)
```

Comment:
I'm not a python expert, but there are two move(c_out_table.tbl) that would look sketchy in c++ -> shouldn't the second one be replaced by 'ctbl' ?

Reply:
Agreed, we shouldn't need the second move here.

Reply:
ctbl shouldn't exist at all. This was something left over from me trying to troubleshoot the segmentation fault and should be removed. I will remove it.